interview-questions.md

```diff
@@ -1,51 +1,62 @@
+
+Researched answer: As mentioned before, yes you need to call super within the initialize method of the child class. But also, super goes in tandem with the declaration of the child class. The parent class must be mentioned, and the syntax looks as below. 
+
+class ChildClass < ParentClass
+end
+
 
-Researched answer:
 
 2. What is a gem?
 
-Your answer:
+Your answer: Gem is a Ruby library that allows for users to perform tests with their code to ensure that it runs properly and as intended. Going on a limb here I hope I'm right but when we say Gem is a Ruby library, that means that it is a collection of files created by developers worldwide, and this library runs behind the scenes to allow the testing to happen in the first place. It's the code that lets the testing happen, the code behind the code.
+
+Researched answer: In addition to my answer prior, it should be noted that gems are managed with RubyGems, a package manager (would you have guessed). Gems also allow for code to be shared between developers from anywhere. Gems to Ruby is comparable to Yarn with JavaScript.
+
 
-Researched answer:
 
 3. What is a relational database? Are there other kinds of databases?
 
-Your answer:
+Your answer: A relational database is an organized database containing data on the entirety of one particular subject, like the countries database that these laptops came installed with. I'm not sure if it comes in other forms, but I know that it at least can come in table format, with rows dictating single entries (the countries) and the columns dictating different points of data for those entries (country population, country surface area, etc.).
+
+Researched answer: Oh yes, relational databases do indeed come in table format (excel but better). Another point to mention though, is the schema, which is the term defining every table and defining point (column headers) within that specific database.
+
 
-Researched answer:
 
 4. What are primary keys? Why are they important?
 
-Your answer:
+Your answer: Primary keys are unique identifiers in relational databases that ensures that every single entry has at least one distinct characteristic about it. So for example, if two identical entries are put down, everything would be the same except for their primary keys. Typically, those primary keys will look like a random jumble of letters and/or numbers, kind of like car license plates.
+
+Researched answer: Additionally, primary keys can be defined as being either explicit or general. Explicit primary keys refer to primary keys pertaining to one specific table, whereas general primary keys can span across different tables with the same properties (like column headers, table names, etc.).
+
 
-Researched answer:
 
 5. What are the HTTP verbs? What is each verb's corresponding CRUD action?
 
-Your answer:
+Your answer: Need to come back with a researched answer.
```

Comment:
I would recommend trying to connect at least one point to something you do know.  For instance, talk about how you have heard about CRUD and if you can, try to talk about it.